### ISSUE TITLE
jlink error: Hash of java.xml differs to expected hash recorded in java.base

### ISSUE THREAD

**Issue by jan-mue** [SAP/SapMachine#1009]
I'm trying to build a custom Java 11 runtime with jlink but get the following error:
```
Error: Hash of java.xml (87bf41b9c573721b1b185dd6bd03ac29f73e7f16e3cc019caf4686b2d0195852) differs to expected hash (dbd74615944b8aaa45b9caa8663f081d6f43cfd636f62bbf1e269272fd241c5d) recorded in java.base
```
We install SapMachine in an Ubuntu 18.04 Docker image using your .deb packages with apt. The installed version is 11.0.13

To reproduce run `jlink --add-modules java.base,java.xml --output /javaruntime`

This issue is probably similar to https://github.com/corretto/corretto-11/issues/96

**Comment by ansteiner** [comment]
Hi,
the issue is specific to .deb package and 11.0.13. It seams to be that during the .deb packaging the jmods are getting another hash. We are investigating this further.
The tar.gz of the SapMachine 11.0.13 do not have this issue. Perhaps you can use this instead of the .deb package. Also the SapMachine 17.0.1 .deb package shouldn't be affected.
Kind regards,
Andreas

**Comment by RealCLanger** [comment]
I analyzed this further. The JDK (java.base module) stores hashes of the jmods which it checks during jlink runtime. It seems the debian package build modifies the jmod files in a way that the hash changes. Probably it's the dh_strip_nondeterminism step - which is not configurable, though, to my knowledge.

Interestingly, the issue does not occur with SapMachine 17, although we use the same mechanism to build the debian package. I debugged this and found out that the fix for https://bugs.openjdk.java.net/browse/JDK-8243666, which came with OpenJDK 15, changed the way how the hashes are computed such that not the binary as a whole is hashed but just its entries with name and content are considered. I will try to get the change backported to OpenJDK 11 to get this issue fixed.
